src/rocdecode/vaapi/vaapi_videodecoder.cpp

```diff
@@ -216,8 +227,20 @@ rocDecStatus VaapiVideoDecoder::CreateSurfaces() {
             ERR("The surface type is not supported");
             return ROCDEC_NOT_SUPPORTED;
     }
+    surf_attribs.push_back(surf_attrib);
+    uint64_t mod_linear = 0;
+    VADRMFormatModifierList modifier_list = {
+        .num_modifiers = 1,
+        .modifiers = &mod_linear,
+    };
+    if (supports_modifiers_) {
+        surf_attrib.type = VASurfaceAttribDRMFormatModifiers;
+        surf_attrib.value.type = VAGenericValueTypePointer;
+        surf_attrib.value.value.p = &modifier_list;
+        surf_attribs.push_back(surf_attrib);
+    }
     CHECK_VAAPI(vaCreateSurfaces(va_display_, surface_format, decoder_create_info_.width,
-        decoder_create_info_.height, va_surface_ids_.data(), va_surface_ids_.size(), &surf_attrib, 1));
+        decoder_create_info_.height, va_surface_ids_.data(), va_surface_ids_.size(), surf_attribs.data(), surf_attribs.size()));
```

Comment:
Which driver version will have Mesa using tiling by default, and for which ASICs? This change needs further internal discussion and testing for clarity.

Reply:
It should be ready for Mesa 25.1 on all ASICs that supports DRM modifiers. Currently that means GFX9+, but without the compute-only ASICs.